src/plaid/plaid.service.ts

```diff
@@ -8,27 +8,30 @@ import plaid, { Account, Client } from 'plaid'
 import { ConfigService } from '@nestjs/config'
 import { IPlaidConfig } from '../config/plaidConfig'
 import { getOrThrow } from '../core/utils'
+import { AsyncLazy } from '../core/utils/Lazy'
 
 @Injectable()
 export class PlaidService {
-  private readonly client: Client
+  private readonly client: AsyncLazy<Client>
   constructor(
     private readonly encryptionService: EncryptionService,
     private readonly configService: ConfigService,
     @InjectRepository(PlaidItem) private readonly plaidSessionRepo: Repository<PlaidItem>,
   ) {
+
     const config = getOrThrow(this.configService.get<IPlaidConfig>('plaid'))
-    this.client = new plaid.Client(
+    this.client = new AsyncLazy<Client>(async () => await new plaid.Client(
```

Comment:
Is plaid.Client actually async?  I think you can do `Lazy<Client>(() => new plaid.client(...`?

Makes calls to get the client cleaner too as you don't need to `await this.client.get()`